### ISSUE TITLE
RetryTask: Failed resource added for retry.

### ISSUE THREAD

**Issue by kvaps** [LINBIT/linstor-server#144]
Hi, Everytime on controller restart I see:
```
06:59:28.332 [Main] INFO  LINSTOR/Controller - SYSTEM - Controller initialized
06:59:43.286 [MainWorkerPool-1] WARN  LINSTOR/Controller - SYSTEM - RetryTask: Failed resource 'pvc-69f9004d-655d-4db7-907e-ebf8013ea801' of node 'm16c14' added for retry.
06:59:52.270 [MainWorkerPool-1] WARN  LINSTOR/Controller - SYSTEM - RetryTask: Failed resource 'pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3' of node 'm9c23' added for retry.
06:59:52.996 [MainWorkerPool-1] WARN  LINSTOR/Controller - SYSTEM - RetryTask: Failed resource 'pvc-69f9004d-655d-4db7-907e-ebf8013ea801' of node 'm9c25' added for retry.
07:00:08.037 [grizzly-http-server-1] INFO  LINSTOR/Controller - SYSTEM - Volume definition with number '0' successfully  created in resource definition ''.
07:00:08.094 [grizzly-http-server-0] INFO  LINSTOR/Controller - SYSTEM - Volume definition with number '0' successfully  created in resource definition ''.
07:00:14.189 [MainWorkerPool-1] WARN  LINSTOR/Controller - SYSTEM - RetryTask: Failed resource 'pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3' of node 'm16c15' added for retry.
07:00:35.467 [MainWorkerPool-1] ERROR LINSTOR/Controller - SYSTEM - Problem of type 'java.lang.NullPointerException' logged to report number 5EBA4923-00000-000000
07:00:35.505 [grizzly-http-server-0] ERROR LINSTOR/Controller - SYSTEM - HTTP 404 Not Found [Report number 5EBA4923-00000-000001]
07:00:37.502 [grizzly-http-server-0] INFO  LINSTOR/Controller - SYSTEM - Volume definition with number '0' successfully  created in resource definition ''.
07:00:37.672 [grizzly-http-server-0] INFO  LINSTOR/Controller - SYSTEM - Volume definition with number '0' successfully  created in resource definition ''.
```

```console
# linstor r l -r pvc-69f9004d-655d-4db7-907e-ebf8013ea801
+---------------------------------------------------------------------------------------------------+
| ResourceName                             | Node   | Port  | Usage  | Conns             |    State |
|===================================================================================================|
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m16c14 | 55119 | Unused | Ok                | UpToDate |
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m9c11  | 55119 | Unused | Ok                | Outdated |
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m9c25  | 55119 |        | Connecting(m9c11) | DELETING |
+---------------------------------------------------------------------------------------------------+

# linstor n l -n m16c14 m9c11 m9c25
+----------------------------------------------------------+
| Node   | NodeType  | Addresses                  | State  |
|==========================================================|
| m9c11  | SATELLITE | 10.36.129.146:3366 (PLAIN) | Online |
| m9c25  | SATELLITE | 10.36.129.160:3366 (PLAIN) | Online |
| m16c14 | SATELLITE | 10.36.130.239:3366 (PLAIN) | Online |
+----------------------------------------------------------+

root@m9c11:~# drbdadm status pvc-69f9004d-655d-4db7-907e-ebf8013ea801
pvc-69f9004d-655d-4db7-907e-ebf8013ea801 role:Secondary
  disk:Outdated
  m16c14 role:Secondary
    peer-disk:UpToDate

root@m9c25:~# drbdadm status pvc-69f9004d-655d-4db7-907e-ebf8013ea801
pvc-69f9004d-655d-4db7-907e-ebf8013ea801 role:Primary
  disk:Diskless
  m16c14 role:Secondary
    peer-disk:UpToDate
  m9c11 connection:Connecting

root@m16c14:~# drbdadm status pvc-69f9004d-655d-4db7-907e-ebf8013ea801
pvc-69f9004d-655d-4db7-907e-ebf8013ea801 role:Secondary
  disk:UpToDate
  m9c11 role:Secondary
    peer-disk:Outdated
  m9c25 role:Primary
    peer-disk:Diskless
```

```console
# linstor r l -r pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3
+---------------------------------------------------------------------------------------------------+
| ResourceName                             | Node   | Port  | Usage  | Conns             |    State |
|===================================================================================================|
| pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | m16c15 | 55118 | Unused | Ok                | UpToDate |
| pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | m6c10  | 55118 | Unused | Ok                | Outdated |
| pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | m9c23  | 55118 |        | Connecting(m6c10) | DELETING |
+---------------------------------------------------------------------------------------------------+

# linstor n l -n m16c15 m6c10 m9c23
+----------------------------------------------------------+
| Node   | NodeType  | Addresses                  | State  |
|==========================================================|
| m6c10  | SATELLITE | 10.36.129.10:3366 (PLAIN)  | Online |
| m9c23  | SATELLITE | 10.36.129.158:3366 (PLAIN) | Online |
| m16c15 | SATELLITE | 10.36.130.240:3366 (PLAIN) | Online |
+----------------------------------------------------------+

root@m16c15:~# drbdadm status pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3
pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 role:Secondary
  disk:UpToDate
  m6c10 role:Secondary
    peer-disk:Outdated
  m9c23 role:Primary
    peer-disk:Diskless

root@m6c10:~# drbdadm status pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3
pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 role:Secondary
  disk:Outdated
  m16c15 role:Secondary
    peer-disk:UpToDate

root@m9c23:~# drbdadm status pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3
'pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3' not defined in your config (for this host).
```

```console
# linstor r l -r pvc-69f9004d-655d-4db7-907e-ebf8013ea801
+---------------------------------------------------------------------------------------------------+
| ResourceName                             | Node   | Port  | Usage  | Conns             |    State |
|===================================================================================================|
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m16c14 | 55119 | Unused | Ok                | UpToDate |
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m9c11  | 55119 | Unused | Ok                | Outdated |
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m9c25  | 55119 |        | Connecting(m9c11) | DELETING |
+---------------------------------------------------------------------------------------------------+

# linstor n l -n m16c14 m9c11 m9c25
+----------------------------------------------------------+
| Node   | NodeType  | Addresses                  | State  |
|==========================================================|
| m9c11  | SATELLITE | 10.36.129.146:3366 (PLAIN) | Online |
| m9c25  | SATELLITE | 10.36.129.160:3366 (PLAIN) | Online |
| m16c14 | SATELLITE | 10.36.130.239:3366 (PLAIN) | Online |
+----------------------------------------------------------+

root@m16c14:~# drbdadm status pvc-69f9004d-655d-4db7-907e-ebf8013ea801
pvc-69f9004d-655d-4db7-907e-ebf8013ea801 role:Secondary
  disk:UpToDate
  m9c11 role:Secondary
    peer-disk:Outdated
  m9c25 role:Primary
    peer-disk:Diskless

root@m9c11:~# drbdadm status pvc-69f9004d-655d-4db7-907e-ebf8013ea801
pvc-69f9004d-655d-4db7-907e-ebf8013ea801 role:Secondary
  disk:Outdated
  m16c14 role:Secondary
    peer-disk:UpToDate

root@m9c25:~# drbdadm status pvc-69f9004d-655d-4db7-907e-ebf8013ea801
pvc-69f9004d-655d-4db7-907e-ebf8013ea801 role:Primary
  disk:Diskless
  m16c14 role:Secondary
    peer-disk:UpToDate
  m9c11 connection:Connecting
```

**Error report:**
[5EBA4923-00000-000000.log](https://github.com/LINBIT/linstor-server/files/4613973/5EBA4923-00000-000000.log)

**Comment by kvaps** [comment]
I guess it's might be connected with https://github.com/LINBIT/linstor-server/issues/106 and https://github.com/LINBIT/linstor-server/issues/130

**Comment by ghernadi** [comment]
At first look this indeed looks like #106. 

Are these new resources or are those resources still broken since #106 was reported? 
What I am trying to figure out - does the issue still occurs on new resources or did we just not properly migrated those old resources?

Regarding the error report. this is is something completely different and has nothing to do with this broken resource states. Will fix that though :)

**Comment by kvaps** [comment]
>Are these new resources or are those resources still broken since #106 was reported?

Unfortunately I can't say you for sure, I've checked our incidents log, and found nothing about them. 

> What I am trying to figure out - does the issue still occurs on new resources or did we just not properly migrated those old resources?

I think #106 is fully solved, but this issue is more interesting, just checked created volumes state, and all of them have replica with no DeviceName :

```
+--------------------------------------------------------------------------------------------------------------------------------------------+
| Node   | Resource                                 | StoragePool          | VolNr | MinorNr | DeviceName    | Allocated | InUse  |    State |
|============================================================================================================================================|
| m16c14 | pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | thindata             |     0 |    1025 | None          | 72.05 GiB | Unused | UpToDate |
| m9c11  | pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | thindata             |     0 |    1025 | /dev/drbd1025 | 82.62 GiB | Unused | Outdated |
| m9c25  | pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | DfltDisklessStorPool |     0 |    1025 | None          |           | InUse  | Diskless |
+--------------------------------------------------------------------------------------------------------------------------------------------+
 kvaps  ~  linstor r l -r pvc-69f9004d-655d-4db7-907e-ebf8013ea801                                                                                                                                                          prod nextcloud 
+---------------------------------------------------------------------------------------------------+
| ResourceName                             | Node   | Port  | Usage  | Conns             |    State |
|===================================================================================================|
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m16c14 | 55119 | Unused | Ok                | UpToDate |
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m9c11  | 55119 | Unused | Ok                | Outdated |
| pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | m9c25  | 55119 |        | Connecting(m9c11) | DELETING |
+---------------------------------------------------------------------------------------------------+
 kvaps  ~                                                                                                                                                                                                                   prod nextcloud 
 kvaps  ~                                                                                                                                                                                                                   prod nextcloud 
 kvaps  ~                                                                                                                                                                                                                   prod nextcloud 
 kvaps  ~  linstor v l -r pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3                                                                                                                                                          prod nextcloud 
+---------------------------------------------------------------------------------------------------------------------------------------------+
| Node   | Resource                                 | StoragePool          | VolNr | MinorNr | DeviceName    |  Allocated | InUse  |    State |
|=============================================================================================================================================|
| m16c15 | pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | thindata             |     0 |    1023 | None          | 607.38 GiB | Unused | UpToDate |
| m6c10  | pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | thindata             |     0 |    1023 | /dev/drbd1023 | 655.20 GiB | Unused | Outdated |
| m9c23  | pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | DfltDisklessStorPool |     0 |    1023 | None          |            | InUse  | Diskless |
+---------------------------------------------------------------------------------------------------------------------------------------------+
 kvaps  ~  linstor v l -r pvc-69f9004d-655d-4db7-907e-ebf8013ea801                                                                                                                                                          prod nextcloud 
+--------------------------------------------------------------------------------------------------------------------------------------------+
| Node   | Resource                                 | StoragePool          | VolNr | MinorNr | DeviceName    | Allocated | InUse  |    State |
|============================================================================================================================================|
| m16c14 | pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | thindata             |     0 |    1025 | None          | 72.05 GiB | Unused | UpToDate |
| m9c11  | pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | thindata             |     0 |    1025 | /dev/drbd1025 | 82.62 GiB | Unused | Outdated |
| m9c25  | pvc-69f9004d-655d-4db7-907e-ebf8013ea801 | DfltDisklessStorPool |     0 |    1025 | None          |           | InUse  | Diskless |
+--------------------------------------------------------------------------------------------------------------------------------------------+
 kvaps  ~  linstor v l -r pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3                                                                                                                                                          prod nextcloud 
+---------------------------------------------------------------------------------------------------------------------------------------------+
| Node   | Resource                                 | StoragePool          | VolNr | MinorNr | DeviceName    |  Allocated | InUse  |    State |
|=============================================================================================================================================|
| m16c15 | pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | thindata             |     0 |    1023 | None          | 607.38 GiB | Unused | UpToDate |
| m6c10  | pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | thindata             |     0 |    1023 | /dev/drbd1023 | 655.20 GiB | Unused | Outdated |
| m9c23  | pvc-c0d8ada5-d2da-4b9f-8b61-7973a3e110d3 | DfltDisklessStorPool |     0 |    1023 | None          |            | InUse  | Diskless |
+---------------------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by ghernadi** [comment]
Still - in this case I would like to ask you to create a new resource-definition, copy the data from these broken resources and delete those broken resources (including their definition) entirely. 

I admit, this is more like "making sure that those resources are not broken since #106", but I also want to make sure that I have some chance to reproduce this issue. I ran a quick test where I tried to delete a primary diskless resource, but linstor refused to do so. Without deleting a primary diskless, I am not sure how else I could reproduce this issue / where I should start looking for the bug...
